app/views/products/_product_form.html.erb

```diff
@@ -24,7 +24,11 @@
           <% end %>
 
           <%= f.form_group :randomize_records do %>
-            <%= f.check_box :randomize_records, label: 'Randomize Records', checked: true, disabled: params[:action] == "edit" %>
+            <%= f.check_box :randomize_records, label: 'Randomize Records', checked: params[:action] == 'edit' ? @product.randomize_records : true, disabled: params[:action] == 'edit' %>
```

Comment:
You should use @product.new_record? instead of params[:action] == 'edit' to detect if we should disable